### DISCUSSION TITLE
Unirest Object mapping question

### DISCUSSION THREAD

**Discussion by elizarrrrr** [Kong/unirest-java#437] [Q&A]
Hi guys, thanks for the library you made, looks really clean.

But I got a question.

I am trying to retrieve data from jira. The response object has following format.

{
   "size": 1,
    "start": 0,
    "limit": 50,
    "values": [
    list of requests.....
    ]
}

Now how can I access this list? Because when I do 
```
.AsObect(new GenericType<List<Book>>() {
                }  

```
I get the following error:
![grafik](https://user-images.githubusercontent.com/36076242/158343638-fdc321c6-e6c4-432d-aad8-dd7e4ebb1182.png)
Looking at this, I would think that i need an object where values is an attribute in it to retrieve the data, but there is for sure a better way, isn't it?

My actual code:
```
    public List<JiraRequestDTO> getOpenRequests() {
        HttpResponse<JiraDTO> httpResponse = Unirest
                .get(endpoint + issuesEndpoint)
                .basicAuth(mail, apiKey)
                .asObject(JiraDTO.class);
        List<JiraRequestDTO> requests = httpResponse.getBody().values;
        return requests;
   
```

**Comment by ryber** [comment]
That error is because the response is not parseable as JSON so I expect the response is not what you think, if you look on the exception you should be able to get the original text, or call it with asString to see. Maybe you need a application/json accept header?

**Reply by elizarrrrr** [reply]
Application header is already added.

That's how the response looks like:
![grafik](https://user-images.githubusercontent.com/36076242/158411722-c36d44c7-40cb-418e-8de7-c2d9ddf8809e.png)

Also if i get is as String it looks like a JSON. I can't send more details if it might help.

**Reply by ryber** [reply]
Ah ok I see, the object isn't a list, it's an object, so the error is because when you deserialize to a ```List``` it expects the first char to be ```[```

**Reply by elizarrrrr** [reply]
But would it be a way to automatically retrieve a data or values array in the response without needing a class for the whole request?

If no, could that theoretically may be a cool feature request? Since a lot of API's are using this semantic to provide data.

**Reply by ryber** [reply]
You can strip JiraRequestDTO down to the bare minimum, just a class with a values property that is an array, you don't neccessarily need any other fields.

I wouldn't consider this a Unirest type of request, Unirest is really working with an internal JSON parser (Gson or Jackson), those are the libraries doing the work on Object Mapping which is a whole giant world of its own.

**Reply by elizarrrrr** [reply]
I agree thanks for helping out!

Another question, I am using it in combination with Spring Boot. Is there a way to configure Unirest in a constructor of a service or a Configuration class, because right now i need to config Unirest before every request, even if it is in the same class/service? Overall do you know any tutorials how Unirest works with the Spring Context.

**Reply by ryber** [reply]
You can create instances of Unirest and configure them, so in spring something like:

```
   @Bean
    public UnirestInstance unirest() {
            UnirestInstance u = Unirest.spawnInstance();
             u.config().doSomething();
           return u;
     }
```

That assumes the bean is a singleton which I think it is by default.

You could also do the same thing in the constructor of a ```@Service``` or ```@Component```` (again probably want to make sure it's the Singleton scope)

**Reply by elizarrrrr** [reply]
This helps, thanks! Wish you a great week
